docker/optimism/optimism.sh

```diff
@@ -12,73 +11,88 @@ CACHE_DIR=$(obtain_cache_dir $0)
+    cd ${OPTIMISM}
+    git checkout ${commit_id}
+}
+
+run_on_op_launcher () {
+    ex=$1
+    docker run -v /var/run/docker.sock:/var/run/docker.sock \
+        -v ${OPTIMISM}:${OPTIMISM} \
+        --network=host optimism-launcher bash -c "${ex}"
+}
+
 down() {
     echo "Shutting down the end-to-end environment"
-    make -C ${OPTIMISM}/ops down
+    run_on_op_launcher "cd ${OPTIMISM} && make devnet-down && make devnet-clean"
+}
+
+build() {
+    DOCKER_BUILDKIT=1 docker build ${ROOT}/docker/optimism -t optimism-launcher
 }
 
 up() {
     echo "Starting the end-to-end environment"
-    docker_compose_file="-f ${OPTIMISM}/ops/docker-compose.yml"
-    # FIXME: Need removal after: https://github.com/beamer-bridge/beamer/issues/1242
-    export DOCKER_TAG_MESSAGE_RELAYER=0.5.25
-    docker compose ${docker_compose_file} up --scale relayer=0 -d
-
-    echo "Wait to make sure all services are up and running"
-    # The current wait-for-sequencer.sh script in the optimism repo
-    # has a bug in that it does not specify the configuration files.
-    # So work around that here.
-    WAIT_FOR_SEQUENCER_SCRIPT=$(mktemp)-wait-for-sequencer.sh
-    sed "s#docker-compose#docker-compose ${docker_compose_file}#" \
-            "${OPTIMISM}/ops/scripts/wait-for-sequencer.sh" > ${WAIT_FOR_SEQUENCER_SCRIPT}
-    sh ${WAIT_FOR_SEQUENCER_SCRIPT}
+    if [ ! -d ${OPTIMISM} ]; then
+        configure_repo
+    fi
+    run_on_op_launcher "cd ${OPTIMISM} && make devnet-up"
 }
 
 create_deployment_config_file() {
-    ADDRESS=$(addresses | jq '.["Proxy__OVM_L1CrossDomainMessenger"]')
+    ADDRESS='"0x6900000000000000000000000000000000000002"'
     sed "s/\${l1_messenger_args}/${ADDRESS}/" \
         ${ROOT}/scripts/deployment/optimism-local-template.json \
         > ${DEPLOYMENT_CONFIG_FILE}
 }
 
 e2e_test() {
-    l2_rpc=http://localhost:8545
+    l2_rpc=http://localhost:9545
     password=""
-    contract_addresses="${CACHE_DIR}/addresses.json"
-    echo Copying contract addresses to $contract_addresses
-    docker exec ops-deployer-1 cat genesis/addresses.json > $contract_addresses
+    
     e2e_test_fill ${DEPLOYMENT_DIR} ${KEYFILE} "${password}" $l2_rpc
     echo Sending Proof
-    e2e_test_op_proof http://localhost:8545 $l2_rpc $CONTRACT_ADDRESSES $PRIVKEY $e2e_test_l2_txhash
+    
+    e2e_test_op_proof http://localhost:8545 $l2_rpc $PRIVKEY $e2e_test_l2_txhash
+    sleep 20
     echo L1 Resolve
-    e2e_test_relayer http://localhost:8545 $l2_rpc $CONTRACT_ADDRESSES $PRIVKEY $e2e_test_l2_txhash
+    timeout 5m bash -c "until ${relayer} relay \
+                                        --l1-rpc-url '$l1_rpc' \
+                                        --l2-relay-to-rpc-url '$l2_rpc' \
+                                        --l2-relay-from-rpc-url '$l2_rpc' \
+                                        --wallet-private-key '$privkey' \
+                                        --l2-transaction-hash '$txhash'; \
+                    do sleep 1s; done"
```

Comment:
Why do this instead of `e2e_test_relayer`?

Reply:
because we dont have any custom network

Reply:
:woozy_face: